sdk/eventhub/Microsoft.Azure.WebJobs.Extensions.EventHubs/src/Listeners/EventHubListener.cs

```diff
@@ -104,6 +111,98 @@ public IScaleMonitor GetMonitor()
             return _scaleMonitor.Value;
         }
 
+        internal class EventProcessorStoredEventsManager : IDisposable
```

Comment:
I think it would be easier to convert this to something specific to one partition and then keep a dictionary of them in the Listener.    That would let you have each manager thingy own a `System.Threading.Timer` which would call `TriggerExecute` as it's callback.   Managing this timer would let you eliminate the explicit monitoring of background tasks for each partition.